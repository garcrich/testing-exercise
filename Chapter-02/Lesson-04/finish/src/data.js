const names = ["Bill", "Jane"];

function wait(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

export async function getName(id) {
  await wait(200);
  return names.length <= id ? null : names[id];
}
