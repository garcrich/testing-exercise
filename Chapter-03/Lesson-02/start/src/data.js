const people = [
  {
    id: 1,
    firstName: "Bill",
    lastName: "Peters",
  },
  {
    id: 2,
    firstName: "Jane",
    lastName: "Sheers",
  },
  {
    id: 3,
    firstName: "Sue",
    lastName: "Peel",
  },
];

function wait(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

export async function getPerson(id) {
  await wait(200);
  const found = people.filter((p) => p.id === id);
  return found.length === 0 ? "Person Not Found" : found[0];
}

export async function searchPeople(criteria) {
  await wait(200);
  const found = people.filter(
    (p) =>
      p.firstName.toLowerCase().indexOf(criteria.toLowerCase()) > -1 ||
      p.lastName.toLowerCase().indexOf(criteria.toLowerCase()) > -1
  );
  return found;
}
