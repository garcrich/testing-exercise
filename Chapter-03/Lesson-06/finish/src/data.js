const people = [
  {
    id: 1,
    firstName: "Bill",
    lastName: "Peters",
  },
  {
    id: 2,
    firstName: "Jane",
    lastName: "Sheers",
  },
];

function wait(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

export function searchPeople(criteria) {
  const found = people.filter(
    (p) =>
      p.firstName.toLowerCase().indexOf(criteria.toLowerCase()) > -1 ||
      p.lastName.toLowerCase().indexOf(criteria.toLowerCase()) > -1
  );
  if (found.length === 0) {
    throw Error("No people found");
  }
  return found;
}

export async function searchPeopleAsync(criteria) {
  await wait(200);
  const found = people.filter(
    (p) =>
      p.firstName.toLowerCase().indexOf(criteria.toLowerCase()) > -1 ||
      p.lastName.toLowerCase().indexOf(criteria.toLowerCase()) > -1
  );
  if (found.length === 0) {
    throw Error("No people found");
  }
  return found;
}
