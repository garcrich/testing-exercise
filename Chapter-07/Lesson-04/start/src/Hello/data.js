export async function getUser(id) {
  const response = await fetch(`https://some-server/users/${id}`);
  const data = await response.json();
  return data;
}
