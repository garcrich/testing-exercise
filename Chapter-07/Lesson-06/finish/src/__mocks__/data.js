export async function getUser(id) {
  return new Promise((resolve) =>
    resolve({
      id,
      name: "Bob",
    })
  );
}

export async function getUserRole(id) {
  return new Promise((resolve) =>
    resolve({
      role: "Administrator",
    })
  );
}
