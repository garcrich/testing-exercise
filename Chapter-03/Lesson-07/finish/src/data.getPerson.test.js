import { getPerson } from "./data";

test("Should return correct object when valid id", async () => {
  const result = await getPerson(1);
  expect(result).toMatchSnapshot();
});
