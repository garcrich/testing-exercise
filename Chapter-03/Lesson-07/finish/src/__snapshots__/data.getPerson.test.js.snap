// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return correct object when valid id 1`] = `
Object {
  "firstName": "Bill",
  "id": 1,
  "lastName": "Peters",
}
`;
